src/realm/set.hpp

```diff
@@ -532,6 +532,9 @@ UpdateStatus Set<T>::init_from_parent(bool allow_create) const
         m_tree.reset(new BPlusTree<T>(get_alloc()));
         const ArrayParent* parent = this;
         m_tree->set_parent(const_cast<ArrayParent*>(parent), 0);
+        if constexpr (realm::is_any_v<T, StringData, Mixed>) {
```

Comment:
This explicit call of `set_interner` looks like `set_parent` … if we forget to call it, we may have weird bugs. Is there a way to limit this? Like, embed the initialization in the creation of the tree itself.

Reply:
Maybe. But I think it would generate some redundant code.